provider/lxd/server.go

```diff
@@ -106,21 +106,36 @@ type CloudSpec struct {
 type ServerFactory interface {
 	// LocalServer creates a new lxd server and augments and wraps the lxd
 	// server, by ensuring sane defaults exist with network, storage.
-	LocalServer() (Server, error)
+	// The LXD project must be specified.
+	LocalServer(project string) (Server, error)
 
 	// LocalServerAddress returns the local servers address from the factory.
 	LocalServerAddress() (string, error)
 
 	// RemoteServer creates a new server that connects to a remote lxd server.
 	// If the cloudSpec endpoint is nil or empty, it will assume that you want
 	// to connection to a local server and will instead use that one.
+	// The LXD project must be specified.
```

Comment:
Te comment as written implies this method takes the project as an arg. Perhaps there's a way to re word it or even leave it out, since the public face of the method itself doesn't really deal with it; it's more about how the client itself has been constructed.